File: app/src/main/java/com/cristhian/gametobeat/network/RetrofitClientInstance.java
package com.cristhian.gametobeat.network;

import com.cristhian.gametobeat.BuildConfig;
import com.cristhian.gametobeat.util.IgdbUtil;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClientInstance {

    private static Retrofit mRetrofit;

    public static Retrofit getInstance() {
        if (mRetrofit == null) {

            OkHttpClient okHttpClient = new OkHttpClient.Builder().addInterceptor(new Interceptor() {
                @Override
                public Response intercept(Chain chain) throws IOException {

                    Request original = chain.request();

                    Request request = original
                            .newBuilder()
                            .addHeader("Authorization", "Bearer "+BuildConfig.IgdbAuthorization)
                            .addHeader("Client-ID", BuildConfig.IgdbKey)
                            .build();

                    return chain.proceed(request);
                }
            }).build();

            mRetrofit = new Retrofit
                    .Builder()
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(okHttpClient)
                    .baseUrl(IgdbUtil.BASE_URL)
                    .build();
        }

        return mRetrofit;
    }


}

File: app/src/main/java/com/cristhian/gametobeat/network/RetrofitClientInstance.kt
package com.cristhian.gametobeat.network

import com.cristhian.gametobeat.BuildConfig
import com.cristhian.gametobeat.util.IgdbUtil
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClientInstance {

    private const val BASE_URL = "https://api-v3.igdb.com/"
    const val BASE_IMAGE_URL = "https://images.igdb.com/igdb/image/upload/t_"
    const val IMAGE_URL_COVER_BIG = "cover_big"

    private val okHttpClient =
            OkHttpClient.Builder().addInterceptor { chain ->
                val original = chain.request()
                val request = original
                        .newBuilder()
                        .addHeader(
                                "user-key",
                                BuildConfig.IgdbKey
                        )
                        .build()
                chain.proceed(request)
            }.build()

    val igdbApiService: IgdbApiService by lazy {
        return@lazy Retrofit.Builder()
                .addConverterFactory(GsonConverterFactory.create())
                .client(okHttpClient)
                .baseUrl(BASE_URL)
                .build()
                .create(IgdbApiService::class.java)
    }
}